src/main/java/com/puppycrawl/tools/checkstyle/checks/UniquePropertiesCheck.java

```diff
@@ -75,12 +75,12 @@ protected void processFiltered(File file, FileText fileText) {
                     ex.getLocalizedMessage());
         }
 
-        for (Entry<String> duplication : properties
+        for (Entry<String, AtomicInteger> duplication : properties
```

Comment:
Why AtomicInteger is used

Reply:
@romani if you look at `AbstractMapBasedMultiset` which is what guava's `HashMultiset` extends, it also uses `AtomicInteger`.
See http://grepcode.com/file/repo1.maven.org/maven2/com.google.guava/guava/r09/com/google/common/collect/AbstractMapBasedMultiset.java#55

Reply:
> The latest version of AbstractMapBasedMultiset doesn't use AtomicInteger

It uses `Count` which looks virtually like AtomicInteger.
https://github.com/google/guava/blob/master/guava/src/com/google/common/collect/Count.java

> Have you checked that replacing it with plain Integer works?

Both ways work, but you cannot modify an `Integer`. To update the map with the new value you would have to create a new `Integer` and put that into the map and replace the old value. With the `AtomicInteger`, I don't have put the value back into the map since I can just update the instance directly.

Reply:
Oh, sorry, not `UniquePropertiesCheck.getDuplicatedKeys()` - rather, `UniquePropertiesCheck.UniqueProperties.getDuplicatedKeys()`.

And it turns out that `UniquePropertiesCheck.UniqueProperties` is private anyway, so my last point about immutability is completely and utterly moot. Silly me! :P 

So no real objections from me anymore. :)

Reply:
all this usages are internal, we can change at any time. Ok to use AtomicInteger.